pkg/ruler/ruler.go

```diff
@@ -903,12 +1012,19 @@ func (r *Ruler) getShardedRules(ctx context.Context, userID string, rulesRequest
 	}
 
 	var (
-		mergedMx sync.Mutex
-		merged   []*GroupStateDesc
+		mtx    sync.Mutex
+		merged []*GroupStateDesc
+		errs   []error
```

Comment:
Do we need `errs` to be a slice of error? It looks like we only use `len(errs)` but not using the error itself.
In this case, is keeping track the count good enough?